src/meson.build

```diff
@@ -34,7 +34,9 @@ endif
 message('dirmonitor_backend: @0@'.format(dirmonitor_backend))
 
 if dirmonitor_backend == 'kqueue'
-    libkqueue_dep = dependency('libkqueue', required : false)
+    libkqueue_dep = dependency('libkqueue', fallback: ['libkqueue', 'libkqueue_dep'],
+        default_options: default_fallback_options + ['default_library=static']
+    )
```

Comment:
changed for debugging purposes.

libkqueue has a compile time option to enable thread safety guarantees which have been very helpful while working on this.